nautobot/cloud/filters.py

```diff
@@ -0,0 +1,41 @@
+from nautobot.core.filters import (
+    NaturalKeyOrPKMultipleChoiceFilter,
+    SearchFilter,
+)
+from nautobot.dcim.models import Manufacturer
+from nautobot.extras.filters import NautobotFilterSet
+from nautobot.extras.models import SecretsGroup
+
+from .models import CloudAccount
+
+
+class CloudAccountFilterSet(NautobotFilterSet):
+    q = SearchFilter(
+        filter_predicates={
+            "name": "icontains",
+            "account_number": "icontains",
+            "provider__name": "icontains",
+        },
+    )
+    provider = NaturalKeyOrPKMultipleChoiceFilter(
+        field_name="provider",
+        queryset=Manufacturer.objects.all(),
+        to_field_name="name",
+        label="Provider (name or ID)",
```

Comment:
Feel like we need *somehow* to indicate to the user that this is a dcim.Manufacturer, not a circuits.Provider.